nautobot/core/testing/api.py

```diff
@@ -370,7 +370,10 @@ def check_expected_slug(self, obj):
                     expected_slug += self.slugify_function(val)
 
             self.assertNotEqual(expected_slug, "")
-            self.assertEqual(obj.slug, expected_slug)
+            if hasattr(obj, "slug"):
```

Comment:
It feels like there's a better approach than checking for attrs on the model but I don't know what that approach should be.

Reply:
With slugs going away this probably isn't worth spending time on

Reply:
Just a note that this intersects with #3439 also making changes to this test. Probably good to keep in sync.